mutiny-core/src/lib.rs

```diff
@@ -124,10 +142,70 @@ impl<S: MutinyStorage> MutinyWallet<S> {
         Ok(())
     }
 
+    /// Starts a background process that will watch for nostr wallet connect events
+    pub async fn start_nostr_wallet_connect(&self, from_node: PublicKey) {
+        let nostr = self.nostr.clone();
+        let nm = self.node_manager.clone();
+        spawn_local(async move {
+            let mut broadcasted_info = false;
+            loop {
+                if nm.stop.load(Ordering::Relaxed) {
+                    utils::sleep(1_000).await;
+                    continue;
+                };
+
+                let client = Client::new(&nostr.primary_key);
+                client
+                    .add_relays(nostr.relays.clone())
+                    .await
+                    .expect("Failed to add relays");
+                client.connect().await;
+                client.subscribe(vec![nostr.create_nwc_filter()]).await;
+
+                // broadcast NWC info event
+                // todo we only need to broadcast on creation
+                if !broadcasted_info {
+                    if let Ok(event) = nostr.create_nwc_info_event() {
+                        if let Err(e) = client.send_event(event).await {
+                            log_warn!(nm.logger, "Error sending NWC info event: {e}");
+                        } else {
+                            broadcasted_info = true;
+                        }
+                    }
+                }
+
+                // handle NWC requests
+                let mut notifications = client.notifications();
+                while let Ok(notification) = notifications.recv().await {
+                    if nm.stop.load(Ordering::Relaxed) {
+                        utils::sleep(1_000).await;
+                        continue;
+                    };
```

Comment:
Actually there's a better way to handle this if you want to reference some socket code: 

```
            loop {
                let mut read_fut = conn_copy.read().fuse();
                let delay_fut = Box::pin(utils::sleep(1_000)).fuse();
                pin_mut!(delay_fut);
                select! {
                    msg_option = read_fut => {
                        // ...
                    }
                    _ = delay_fut => {
                        if stop_copy.load(Ordering::Relaxed) {
                            break;
                        }
                    }
                }

```

Reply:
Switched it to use that, not entirely sure if I did it correctly.